crates/resolver/src/caching_client.rs

```diff
@@ -118,6 +89,7 @@ where
         options: DnsRequestOptions,
         mut client: Self,
         preserved_records: Vec<(Record, u32)>,
+        depth: u8,
```

Comment:
How do you feel about keeping (a simplified version of) the `DepthTracker` wrapper type so that we have a bit more type-safety for this? Might be overkill here but there's not that much overhead IMO.

See for example https://github.com/djc/askama/pull/866/files#diff-49a796685d9a80124bee5bc4d4a1e85b7ac00a053d745334a758f3a11d6a7d19R367.